mailbox/api/src/test/java/org/apache/james/mailbox/events/KeyContract.java

```diff
@@ -163,6 +163,18 @@ default void unregisterShouldRemoveDoubleRegisteredListener() throws Exception {
                 .event(any());
         }
 
+        @Test
```

Comment:
base on IMAP session consistency, please add test in the case Event be dispatched from other event bus.
```java
       default void registerShouldNotDispatchPastEvents() throws Exception {
            MailboxListener listener = newListener();

             eventBus2().dispatch(EVENT, ImmutableSet.of(KEY_1)).block();

             eventBus().register(listener, KEY_1);

             verify(listener, after(FIVE_HUNDRED_MS).never())
                .event(any());
        }
```

Reply:
I think I will do it in my PR

Reply:
Will add it here